src/virtual-machine.js

```diff
@@ -205,8 +228,25 @@ class VirtualMachine extends EventEmitter {
      * @returns {string} Project in a Scratch 3.0 JSON representation.
      */
     saveProjectSb3 () {
-        // @todo: Handle other formats, e.g., Scratch 1.4, Scratch 2.0.
-        return this.toJSON();
+        const soundDescs = serializeSounds(this.runtime);
+        const costumeDescs = serializeCostumes(this.runtime);
+        const projectJson = this.toJSON();
+
+        const zip = new JSZip();
+
+        // Put everything in a zip file
+        // TODO compression?
```

Comment:
Does JSZip default to no compression? I would expect this to default to medium-strength compression settings or something like that...

Reply:
It defaults to no compresssion, but it's easy to add an argument, and I think you can provide a number (from 1-9) representing a speed/space trade-off:

https://stuk.github.io/jszip/documentation/api_jszip/generate_async.html#compression-and-compressionoptions-options